[Glenn Ng Jun Jie] iP

glenngnng opened the pull request:
# **Gman (Duke) Task Manager**

- Track tasks (Todo, Deadline, Event)
- Text-based
- Easy _as heck_ to learn

## All you need to do is:
1. Download it from [here](https://nus-cs2103-ay2324s1.github.io/website/schedule/week4/project.html).
2. Double-click it.
3. Add your tasks.
4. Let it manage your tasks for you 😉

## Features
- [x] Add and remove tasks.
- [x] Mark and unmark tasks as done/undone.
- [x] Save/read tasks for future updates.
- [x] List out tasks.
- [x] Find tasks with specific keyword.
- [ ] Reminders! (Coming soon!)
- [ ] Future updates!

##  Some basic commands are:
> todo have fun
> deadline have fun /by 2023-10-12
> event iP presentation /from 2023-10-12 /to 2023-11-12
> mark 1
> list
> find have fun
> delete 2

## Here's the main method:
```java
public static void main(String[] args) throws GmanException {
        Ui.greet();
        String userInput = scanner.nextLine();
        Parser.readInput(userInput, taskList);
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

glenngnng committed:
Ranamed, added greeting and exit.

glenngnng committed:
Made Gman echo

glenngnng committed:
Added ability to store whatever text entered by user and display them back to user when requested.

glenngnng committed:
Level 3: added task class and mark as done and unmark feature.

glenngnng committed:
Made subclasses for Task: deadline, event, todo, generictask

glenngnng committed:
added tag

glenngnng committed:
Added some exceptions as per requirements, will add more later

glenngnng committed:
added delete functionality, used arraylist instead of static array

glenngnng committed:
added test

glenngnng committed:
committing one moer time to refresh online IP tracker

glenngnng committed:
Incremented Level-7

glenngnng committed:
Testing branch commit

glenngnng committed:
Testing Merge commit

glenngnng committed:
Merge branch 'branch-Level-7'

glenngnng committed:
added very simple dateTime to deadline

glenngnng committed:
Merge branch 'branch-Level-8'

glenngnng committed:
More OOP

glenngnng committed:
Merge branch 'A-MoreOOP'

glenngnng committed:
refactored code into package duke

glenngnng committed:
Merge branch 'A-Packages'

glenngnng committed:
Merge remote-tracking branch 'origin/add-gradle-support'

glenngnng committed:
changed package name from Duke to duke, added tests

glenngnng committed:
Merge branch 'A-JUnit'

glenngnng committed:
added A-Gradle and A-Jar tag, renamed branches

glenngnng committed:
Add Level-9 find capabilities

glenngnng committed:
Add JavaDocs for deadline Event todo

glenngnng committed:
Merge branch 'branch-Level-9'

glenngnng committed:
Merge branch 'branch-A-JavaDoc'

teoks0199 commented on src/main/java/Duke/Parser.java:
```diff
@@ -0,0 +1,91 @@
+import java.util.Scanner;
+public class Parser {
+
+    private static final String exitWord = "bye";
```
I like this constant declaration for your exit command!

teoks0199 commented on src/main/java/Duke/Parser.java:
```diff
@@ -0,0 +1,91 @@
+    private static Scanner scanner = new Scanner(System.in);
+    public static void readInput(String userInput, TaskList taskList) throws GmanException {
+        while(!userInput.equals(exitWord)) {
+            String[] words = userInput.split(" ");
```
Perhaps a more descriptive name could be used here?

teoks0199 commented on src/main/java/Duke/Storage.java:
```diff
@@ -0,0 +1,49 @@
+        try {
+            File file = new File(FILE_PATH);
+            Scanner fileScanner = new Scanner(file);
+            ArrayList<Task> tasksFromFile = new ArrayList<>();
```
I like this name, it is very clear what it is.

teoks0199 commented on src/main/java/Duke/TaskList.java:
```diff
@@ -0,0 +1,72 @@
+        Storage.writeTasks(tasks);
+    }
+
+    public void findTasks(String keyword) {
```
I like this name as it shows that more than one task may be found!

teoks0199 reviewed (commented):
Good job!

glenngnng committed:
Add more javadocs and improve coding standard.

McNaBry commented on src/main/java/Duke/TaskList.java:
```diff
@@ -0,0 +1,122 @@
+    public void deleteTask(int index) throws GmanException{
+        try {
+            if (index > this.getSize() || this.getSize() == 0) {
+                throw new GmanException("HEYHEYHEY! There's nothing to delete here!");
```
I like the flavor you give to your exception messages 😆

McNaBry commented on src/main/java/Duke/Event.java:
```diff
@@ -0,0 +1,65 @@
+    @Override
+    public String toString() {
+        return ("[E]" + super.getStatusIcon() + description +
+                " (from: " + from + "to: " + to + ")");
```
With regards to coding standard, would it be better if you start the new line using the + operator?

McNaBry commented on src/main/java/Duke/Event.java:
```diff
@@ -0,0 +1,65 @@
+        String description = segments[2];
+        String from = segments[3];
+        String to = segments[4];
+        Event toReturn =  new Event(description, from, to);
```
I see some instances where you have double spacing after the equals. Unless you need it for better readability of your code, would it be better if its changed to single space?

McNaBry commented on src/main/java/Duke/Deadline.java:
```diff
@@ -0,0 +1,79 @@
+    public static Deadline readFromFile(String[] segments){
+        String symbol = segments[1];
+        String description = segments[2];
+        String by = segments[3];
```
I like that each part of the file string is given their own variable naming before passing it into the Deadline constructor. Helps with the readability 😄

McNaBry reviewed (commented):
Good job with adding flavor to the chatbot! I also like the name Gman that you gave to your bot!

glenngnng committed:
Add GUI capabilities

glenngnng committed:
Merge branch 'branch-Level-10'

glenngnng committed:
Update build.gradle

glenngnng committed:
Add assert statements

Checks correctness of symbol when task is marked or unmarked.

glenngnng committed:
Add undo capabilities

Undo only applies for the last command written by the user.
Undo only applies for 4 commands: delete, add task(todo, deadline,
event), unmark and mark, since list, find, and other commands
does not edit the task list.

glenngnng committed:
Merge pull request #1 from glenngnng/branch-A-Assertions

Add assert statements

glenngnng committed:
Merge branch 'branch-A-Assertions'

glenngnng committed:
Merge branch 'branch-C-Undo'

# Conflicts:
#	data/gman.txt
#	src/main/java/Duke/Parser.java

glenngnng committed:
Merge branch 'master' of https://github.com/glenngnng/ip

glenngnng committed:
Resolve merge conflict in master

glenngnng committed:
Remove commented out code

glenngnng committed:
Fix GUI not greeting

glenngnng committed:
Change package name to gman

Somehow, my entire thing broke so I had to do this.
All this time, my package name was duke and my dir name was Duke,
and the program ran fine.

glenngnng committed:
Fix DateTime for Deadline, add package for tests

glenngnng committed:
Change code structure for better code quality

Shortened method in Parser using abstraction.
Minor changes to other classes for readability.

glenngnng committed:
Merge branch 'branch-changes'

glenngnng committed:
Change exception output from text to GUI.

Exceptions for marking/unmarking a task that is already
marked/unmarked.

glenngnng committed:
Move abstractions around for readability and add Javadocs for Parser

glenngnng committed:
Merge branch 'branch-changes'

glenngnng committed:
Change GUI to be longer in height, add UI.png

glenngnng committed:
Name Gman GUI, change Ui screenshot

glenngnng committed:
Add JavaDocs, add new JUnit tests, improve JUnit tests, Code Quality changes

glenngnng committed:
Add user guide

glenngnng committed:
Add more features to readme

glenngnng committed:
Fix some issues in readme user guide

glenngnng committed:
clean up comments, use Egyptian Bracing for one case

glenngnng committed:
Change name from Ui.png.png to Ui.png

glenngnng committed:
Change mainClass in build.gradle, set archiveFileName as gman.jar

glenngnng committed:
Change FXML file to fit GUI window for smaller screens.

glenngnng committed:
Fix spelling typos
